tests/integration/test_sort.py

```diff
@@ -301,11 +295,60 @@ def test_sort_strings(c, gpu):
     assert_eq(df_result, df_expected, check_index=False)
 
 
-@pytest.mark.skip(reason="WIP DataFusion")
 @pytest.mark.parametrize("gpu", [False, pytest.param(True, marks=pytest.mark.gpu)])
 def test_sort_not_allowed(c, gpu):
     table_name = "gpu_user_table_1" if gpu else "user_table_1"
 
     # Wrong column
     with pytest.raises(Exception):
         c.sql(f"SELECT * FROM {table_name} ORDER BY 42")
+
+
+@pytest.mark.xfail(Reason="Projection step before sort currently failing")
+@pytest.mark.parametrize(
+    "input_table_1",
+    ["user_table_1", pytest.param("gpu_user_table_1", marks=pytest.mark.gpu)],
+)
+def test_sort_by_old_alias(c, input_table_1, request):
+    user_table_1 = request.getfixturevalue(input_table_1)
+
+    df_result = c.sql(
+        f"""
+    SELECT
+        b AS my_column
```

Comment:
Does this specifically fail when we attempt to `SELECT AS`, or would this also be the case if we just did a standard `SELECT` here?

Reply:
it specifically fails when I do a `SELECT old_name as new_name ORDER BY old_name` while the case `SELECT old_name as new_name ORDER BY new_name` works fine. 

There is a small difference in the plans generated for both cases and the error actually originates from a projection step before it even get's to sort. I've added this test in for this edge case but I'll open up an issue to explore the errors in the context of Projection and not sort.